demos/http/http_demo_s3_generate_presigned_url/http_demo_s3_generate_presigned_url.c

```diff
@@ -0,0 +1,617 @@
+
+/**
+ * @brief Generate and print a pre-signed URL to the S3 object file that is specified in pPath.
+ *
+ *
+ * @param[in] pHost The server host address.
+ * @param[in] hostLen The length of the server host address.
+ * @param[in] pPath The Request-URI to the objects of interest. This string
+ * should be null-terminated.
+ *
+ * @return The status of the pre-signed URL generation: true on success, false on failure.
+ */
+static bool printS3ObjectFilePresignedURL( const char * pHost,
+                                           size_t hostLen,
+                                           const char * pPath );
+
+/**
+ * @brief CryptoInterface provided to SigV4 library for generating the hash digest.
+ */
+static SigV4CryptoInterface_t cryptoInterface =
+{
+    .hashInit      = sha256Init,
+    .hashUpdate    = sha256Update,
+    .hashFinal     = sha256Final,
+    .pHashContext  = &hashContext,
+    .hashBlockLen  = HEX_ENCODED_SHA256_HASH_DIGEST_LENGTH,
+    .hashDigestLen = SHA256_HASH_DIGEST_LENGTH,
+};
+
+/**
+ * @brief SigV4 parameters provided to SigV4 library by the application for generating
+ * the Authorization header.
+ */
+static SigV4Parameters_t sigv4Params =
+{
+    .pCredentials     = &sigvCreds,
+    .pDateIso8601     = pDateISO8601,
+    .pRegion          = AWS_S3_BUCKET_REGION,
+    .regionLen        = sizeof( AWS_S3_BUCKET_REGION ) - 1,
+    .pService         = AWS_S3_SERVICE_NAME,
+    .serviceLen       = sizeof( AWS_S3_SERVICE_NAME ) - 1,
+    .pCryptoInterface = &cryptoInterface,
+    .pHttpParameters  = NULL
+};
+
+/*-----------------------------------------------------------*/
+
+static bool generateS3ObjectFilePresignedURL( const char * pPath )
+{
+    bool returnStatus = false;
+
+    assert( pPath != NULL );
+
+    /* Initialize all HTTP Client library API structs to 0. */
+    ( void ) memset( &requestHeaders, 0, sizeof( requestHeaders ) );
+    ( void ) memset( &requestInfo, 0, sizeof( requestInfo ) );
+    ( void ) memset( &response, 0, sizeof( response ) );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = serverHost;
+    requestInfo.hostLen = serverHostLength;
+    requestInfo.pMethod = HTTP_METHOD_GET;
+    requestInfo.methodLen = HTTP_METHOD_GET_LENGTH;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. This is done in
+     * order to download the file in parts. */
+    requestInfo.reqFlags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    /* Initialize the response object. The same buffer used for storing request
+     * headers is reused here. */
+    response.pBuffer = userBuffer;
+    response.bufferLen = USER_BUFFER_LENGTH;
```

Comment:
Did a bit of testing and found all of these lines (and their corresponding globals) to be unneeded. Could you remove these?